lib/src/linting/handle-lint-violations.spec.ts

```diff
@@ -0,0 +1,220 @@
+import { handleLintViolations } from "./handle-lint-violations";
+
+describe("handle lint violations", () => {
+  let deferExit: boolean;
+  let errorMock: jest.Mock;
+  let warnMock: jest.Mock;
+
+  describe("lint violation when rule not specified", () => {
+    beforeEach(() => {
+      errorMock = jest.fn();
+      warnMock = jest.fn();
+
+      const groupedLintErrors = [
+        {
+          name: "error",
+          violations: [{ message: "error_msg" }]
+        }
+      ];
+
+      const spotConfig = {
+        rules: {}
+      };
+
+      deferExit = handleLintViolations(groupedLintErrors, spotConfig, {
+        error: errorMock,
+        warn: warnMock
+      });
+    });
+
+    it("should return deferExit true", () => {
```

Comment:
Personally I'd group these three together and include the setup code inside the test, renaming the overall thing to `it("triggers an error when a rule is not specified")`.

Something like:
```ts
beforeEach(() => {
  errorMock = jest.fn();
  warnMock = jest.fn();
});

it("triggers an error when a rule is not specified", () => {
  ...
  expect(deferExit).toBe(true);
  expect(errorMock).toBeCalled();
  expect(warnMock).not.toBeCalled();
});

..
```

Up to you though, we're in the realm of personal preference.